src/internal/connector/exchange/attachment.go

```diff
@@ -44,9 +44,18 @@ func uploadAttachment(
 	attachment models.Attachmentable,
 ) error {
 	logger.Ctx(ctx).Debugf("uploading attachment with size %d", *attachment.GetSize())
+	attachmentType := attachmentType(attachment)
+
+	// Reference attachments that are inline() do not need to be recreated. The contents are part of the body.
+	if attachmentType == models.REFERENCE_ATTACHMENTTYPE &&
+		attachment.GetIsInline() != nil && *attachment.GetIsInline() {
+		logger.Ctx(ctx).Debugf("skip uploading inline reference attachment: ", *attachment.GetName())
+		return nil
+	}
```

Comment:
The only gotcha is if for whatever reason the email returned as text rather than HTML. This has never been an issue; however, that is a time that the above statement would not be true.